homeassistant/components/zimi/config_flow.py

```diff
@@ -0,0 +1,186 @@
+"""Config flow for zcc integration."""
+
+from __future__ import annotations
+
+import contextlib
+import logging
+import socket
+from typing import Any
+
+import voluptuous as vol
+from zcc import (
+    ControlPoint,
+    ControlPointDescription,
+    ControlPointDiscoveryService,
+    ControlPointError,
+)
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_PORT
+from homeassistant.helpers.device_registry import format_mac
+
+from . import async_connect_to_controller
+from .const import DOMAIN, TITLE
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=""): str,
+        vol.Required(CONF_PORT, default=5003): int,
+        vol.Optional(CONF_MAC, default=""): str,
+    }
+)
+
+
+def _error_tuple(base: str, error_detail: Any) -> tuple[dict[str, str], dict[str, str]]:
+    """Build an error and description tuple."""
+    if not error_detail:
+        return ({"base": base}, {})
+    return ({"base": base}, {"error_detail": str(error_detail)})
+
+
+class ZimiConfigException(Exception):
+    """Base class for ZimiConfig exceptions."""
+
+    CANNOT_CONNECT = "cannot_connect"
+    CONNECTION_REFUSED = "connection_refused"
+    DISCOVERY_FAILURE = "discovery_failure"
+    INVALID_HOST = "invalid_host"
+    INVALID_MAC = "invalid_mac"
+    MISMATCHED_MAC = "mismatched_mac"
+    TIMEOUT = "timeout"
+    UNKNOWN = "unknown"
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Initialise ZimiConfigException allowing passing of error_detail."""
+        super().__init__(*args)
+        self.error_base = args[0]
+        try:
+            self.error_detail = args[1]
+        except IndexError:
+            self.error_detail = None
+
+
+class ZimiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for zcc."""
+
+    api: ControlPoint | None = None
```

Comment:
```suggestion
    api: ControlPoint
```
why the None default?